### PR TITLE
feat (worker): download artifact through swift temp url

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/artifact/artifact.go [hunk 1]
```diff
@@ -82,13 +101,56 @@ func LoadArtifactsByBuildNumber(db gorp.SqlExecutor, pipelineID int64, applicati
 		}
 		arts = append(arts, art)
 	}
+
+	getFetchURL(arts)
+
+	log.Info("Artifacts %+v", arts)
```
Comment (yesnault) [inline engine/api/artifact/artifact.go]: LoadArtifactsByBuildNumber>

File: engine/api/objectstore/swift.go [hunk 2]
```diff
@@ -167,5 +168,8 @@ func (s *SwiftStore) FetchURL(o Object) (string, string, error) {
 	}
 
 	url := s.ObjectTempUrl(container, object, string(key), "GET", time.Now().Add(15*time.Minute))
+
+	log.Info("Fetch URL: %s", string(url))
```
Comment (yesnault) [inline engine/api/objectstore/swift.go]: FetchURL>

File: sdk/artifact.go [hunk 3]
```diff
@@ -91,45 +90,82 @@ func DownloadArtifacts(project string, application string, pipeline string, tag
 
 func download(project, app, pip string, a Artifact, destdir string) error {
 	var lasterr error
+	var reader io.ReadCloser
+	var doRequest func() (io.ReadCloser, int, error)
+
+	if a.TempURL != "" && a.TempURLSecretKey != "" {
+		if verbose {
+			fmt.Printf(">>> downloading artifact %s from %s\n", a.Name, a.TempURL)
```
Comment (yesnault) [inline sdk/artifact.go]: sdk.log

File: sdk/artifact.go [hunk 4]
```diff
@@ -91,45 +90,82 @@ func DownloadArtifacts(project string, application string, pipeline string, tag
...
+	if err != nil {
+		return err
+	}
 
-		f.Close()
-		if err == nil {
-			return nil
-		}
+	//It panics
+	_, err = io.Copy(f, reader)
```
Comment (yesnault) [inline sdk/artifact.go]: online

File: sdk/artifact.go [hunk 5]
```diff
@@ -91,45 +90,82 @@ func DownloadArtifacts(project string, application string, pipeline string, tag
...
+	if err != nil {
+		return err
+	}
+
+	f.Close()
+	if err == nil {
```
Comment (yesnault) [inline sdk/artifact.go]: no need to check here, already checked

File: engine/api/objectstore/swift.go (hunk 2 above)
Comment (sguiheux) [inline engine/api/objectstore/swift.go]: Debug?

File: sdk/artifact.go [hunk 6]
```diff
@@ -91,45 +90,82 @@ func DownloadArtifacts(project string, application string, pipeline string, tag
...
+		return nil
 	}
 
 	return fmt.Errorf("x5: %s", lasterr)
```
Comment (yesnault) [inline sdk/artifact.go]: not reacheable code, no?

File: sdk/artifact.go [hunk 7]
```diff
@@ -287,7 +322,9 @@ func uploadArtifactWithTempURL(project, pipeline, application, env, tag string,
 		return err
 	}
 
-	fmt.Println("Temprary URL: ", art.TempURL)
+	if verbose {
+		fmt.Printf("Uploading %s with to %s", art.Name, art.TempURL)
```
Comment (yesnault) [inline sdk/artifact.go]: sdk.log

File: engine/worker/take_pipeline_build_job.go [hunk 8]
```diff
@@ -68,7 +68,7 @@ func (w *currentWorker) takePipelineBuildJob(ctx context.Context, pipelineBuildJ
 		for {
 			select {
 			case <-ctx.Done():
-				log.Error("Exiting pipeline build job info goroutine: %v", ctx.Err())
+				log.Info("Exiting pipeline build job info goroutine: %v", ctx.Err())
```
Comment (sguiheux) [inline engine/worker/take_pipeline_build_job.go]: debug?

**Pull request merged**